integrade/tests/conftest.py

```diff
@@ -35,3 +38,25 @@ def drop_account_data():
     else:
         pytest.skip('Must be able to drop account data for this test to work.'
                     'Make sure the "oc" client is in your PATH.')
+
+
+@pytest.fixture()
+def instances_to_terminate():
+    """Provide list to test to indicate instances that should be terminated.
+
+    We must know what aws profile to use, so append tuples of (aws_profile,
+    instance_id) to the list.
+
+    The cleanup code will run after the test even if it fails, so instance ids
+    should be added to the list immediately after creation, so if something
+    fails, they can be cleaned up.
+    """
+    instances_to_terminate = []
+
+    yield instances_to_terminate
+
+    if instances_to_terminate:
+        num_instances = len(instances_to_terminate)
+        with Pool(num_instances) as p:
+            p.map(
+                terminate_instance, instances_to_terminate)
```

Comment:
Maybe join could be used whenever a Pool was created to make sure the all processes complete before moving on.

https://docs.python.org/3.5/library/multiprocessing.html#multiprocessing.pool.Pool.join

Reply:
Reading https://docs.python.org/2/library/multiprocessing.html#multiprocessing.pool.multiprocessing.Pool.map I understood "blocks until it finishes" to mean no additional join was needed

Reply:
I've been thinking a little bit more about using the close and join and I checked the Pool's source code and realized that with the context manager it will terminate (see https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/multiprocessing/pool.py#L574-L575).

That said and considering that the `map` will only return after completing all tasks calling close and join is useless as you pointed. I saw that you've added the close and join so feel free to remove it. I apologize for that, when I was reviewing I didn't had the fact that you were using Pool as a context manager.